docs/docsite/rst/playbook_guide/playbook_pathing.rst

```diff
@@ -21,25 +21,30 @@ Task paths
 
-Resolving local relative paths
-------------------------------
+How plugins resolve local relative paths
+----------------------------------------
 
 When you specify a relative path for a local file, Ansible will try to find that file first in the current task's role, then in other roles that included or depend on the current role, then relative to the file in which the task is defined, and finally relative to the current play. It will take the first matching file that it finds. This way, if multiple files with the same file name exist, Ansible will find the file that is closest to the current task and that is most likely to be file you wanted.
 
-Specifically, Ansible tries to find the file
+Specifically, Ansible tries to find the file in the following order:
 
 1. In the current role.
 
    1. In its appropriate subdirectory—"files", "vars", "templates" or "tasks", depending on the kind of file Ansible is searching for.
    2. Directly in its directory.
    
-2. Like 1, in the parent role that called into this current role with `include_role`, `import_role`, or with a role dependency. If the parent role has its own parent role, Ansible will repeat this step with that role.
+2. Like 1, in the parent role that called into this current role with ``include_role``, ``import_role``, or with a role dependency. If the parent role has its own parent role, Ansible will repeat this step with that role.
 3. Like 1, in the current task file's directory.
 4. Like 1, in the current play file's directory.
 
-Ansible does not search the current working directory. (The directory you're in when you execute Ansible.) Also, Ansible will only search within a role if you actually included it with an `include_role` or `import_role` task or a dependency. If you instead use `include`, `include_task` or `import_task` to include just the tasks from a specific file but not the full role, Ansible will not search that role in steps 1 and 2.
+Ansible does not search for local files in the current working directory, in other words, the directory from which you execute Ansible.
 
-When you execute Ansible, the variable `ansible_search_path` will contain the paths searched, in the order they were searched in but without listing their subdirectories. If you run Ansible in verbosity level 5 by passing the `-vvvvv` argument, Ansible will report each directory as it searches, except when it searches for a tasks file.
+.. note:: When resolving local relative paths for tasks files, Ansible uses the context of the role that includes tasks with an ``include_role`` or ``import_role`` statement. If you import the tasks with ``include_task``, or ``import_task`` statements, Ansible uses the context of the importing file.
```

Comment:
```suggestion
.. note:: When resolving local relative paths for files in tasks, Ansible uses the context of the role that includes tasks with an ``include_role`` or ``import_role`` statement. If you import the tasks with ``include_task``, or ``import_task`` statements, Ansible uses the context of the importing file.
```

Reply:
it is just not 'use' but adds it to the stack, with highest precedence

Reply:
you can just have a list of notes in one  `.. note::` section, less boxy

Reply:
ansible 'adds' the role to the search context, does not ONLY use the role as the current text implies, the search context is a stack and the wording should reflect that to avoid confusion. 

A common issue that comes up is 'why did ansible take the file from the parent role/play/etc when i misspelled the file in the role?'  this is born of a misunderstanding that presumes the role is the ONLY context in the search and not the whole stack. Why this should be very clear, context is ADDITIVE when it comes to search paths.